### ISSUE TITLE
Support for variable length fraction seconds

### ISSUE THREAD

**Issue by prymitive** [freach/udatetime#15]
Currently only 3 and 6 digit fraction seconds are supported:

```time-secfrac can be either 3DIGIT for milliseconds or 6DIGIT for microseconds```

So 2 digit ones fail:

```
In [1]: import udatetime

In [2]: udatetime.from_string("2016-07-15T12:33:20.123000+02:00")
Out[2]: datetime.datetime(2016, 7, 15, 12, 33, 20, 123000, tzinfo=+02:00)

In [3]: udatetime.from_string("2016-07-15T12:33:20.123000Z")
Out[3]: datetime.datetime(2016, 7, 15, 12, 33, 20, 123000, tzinfo=+00:00)

In [4]: udatetime.from_string("2016-07-15T12:33:20.12Z")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-240aac401a4c> in <module>()
----> 1 udatetime.from_string("2016-07-15T12:33:20.12Z")

ValueError: Invalid RFC3339 date-time string. Time invalid.
```

I don't control data source so to use udatetime I would need to do string conversion which might eat all performance benefits.
Would it be possible to support any fraction seconds up to  6 digits?

**Comment by tdamsma** [comment]
Indeed the number of fractional digits is optional according to https://tools.ietf.org/html/rfc3339:

> time-secfrac    = "." 1*DIGIT

**Comment by freach** [comment]
Feature implemented. Will be released in 0.0.16.
